beacon_chain/utils/ssz/src/encode.rs

```diff
@@ -83,6 +80,7 @@ pub fn encode_length(len: usize, length_bytes: usize) -> Vec<u8> {
     assert!(length_bytes > 0);  // For sanity
     assert!((len as usize) < 2usize.pow(length_bytes as u32 * 8));
     let mut header: Vec<u8> = vec![0; length_bytes];
+    #[allow(needless_range_loop)]
```

Comment:
I saw the clippy lint for this was a bit awkward, here's a solution that'll please clippy whilst still being readable:

```rust
pub fn encode_length(len: usize, length_bytes: usize) -> Vec<u8> {
    assert!(length_bytes > 0);  // For sanity
    assert!((len as usize) < 2usize.pow(length_bytes as u32 * 8));
    let mut header: Vec<u8> = vec![0; length_bytes];
    for (i, header_byte) in header.iter_mut().enumerate() {
        let offset = (length_bytes - i - 1) * 8;
        *header_byte = ((len >> offset) & 0xff) as u8;
    };
    header
}
```

Thoughts?

Reply:
Ah, I see. I wasn't dereferencing `header_byte` when I tried using the mutable iterator. Thanks for getting this right. Changes are committed.